### PR TITLE
Allow changing shade slot cap + prefer maintaining data

### PR DESCRIPTION
**Pull request by SupersonicNK**
Shade slots above the normal cap (8) are still readable/writable in rivals, but aren't used by the color shader. You can however use them to store miscellaneous data. Reading them is done using the functions `get_profile_color_slot_r/g/b`
For example, I use it to store the two endpoints of a color transition in an upcoming character.

I also modified the parsing functions to increase the maximum slot values rather than destroy the additional colors and shades, as it makes for less steps when importing a colors.gml file with more than the normal amount of data, because that additional data is still usable in-game.

additionally, a warning was added to color and shade slots above the game's technical maximum.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -448,8 +450,11 @@ const vm = new Vue({
 			})
 
 			//only keep 16 color profiles
+			//if (this.colorProfilesMainColors.length > this.MAX_ALT_PALETTES)
+			//	this.colorProfilesMainColors.splice(this.MAX_ALT_PALETTES, this.colorProfilesMainColors.length - 1)
+
 			if (this.colorProfilesMainColors.length > this.MAX_ALT_PALETTES)
-				this.colorProfilesMainColors.splice(this.MAX_ALT_PALETTES, this.colorProfilesMainColors.length - 1)
+				this.MAX_ALT_PALETTES = this.colorProfilesMainColors.length; //prefer maintaining user data.
```
Comment (CL-9a) [inline index.js]: also not sure this comment's really useful in the context of a comment in the code - since it says "prefer" it refers to the change you've made in the commit, but a better place to do that would be to do it in the PR or in the commit message (and you did both already :muscle:)

(sorry, that's kinda nitpicking haha, but since I'm already asking to just clean up the commented code, I might as well!!)

File: index.js [hunk 2]
```diff
@@ -470,7 +475,10 @@ const vm = new Vue({
 			this.rows = [];
 
 			try {
-				this.rows = json.splice(0, this.MAX_SHADE_ROWS);
+				//this.rows = json.splice(0, this.MAX_SHADE_ROWS);
```
Comment (CL-9a) [inline index.js]: same here, can you remove old code?

File: index.js [hunk 3]
```diff
@@ -448,8 +450,11 @@ const vm = new Vue({
 			})
 
 			//only keep 16 color profiles
+			//if (this.colorProfilesMainColors.length > this.MAX_ALT_PALETTES)
+			//	this.colorProfilesMainColors.splice(this.MAX_ALT_PALETTES, this.colorProfilesMainColors.length - 1)
+
```
Comment (CL-9a) [inline index.js]: Could you just straight up remove the old code instead of commenting it? It's tracked in git anyway so it's not lost, and it makes it easier to see why stuff is here

File: index.js [hunk 4]
```diff
@@ -879,6 +889,54 @@ const vm = new Vue({
...
+									cachedColorTransforms.set(`${r},${g},${b}`, shiftedRgb2);
+
+									//console.log("px", i, "fitting rangeDef", hsv, mainColorForShade.hsv, defaultToCurrentDeltaHSV, shiftedRgb)
+								}
+								
+								shadeIndex++;
+							}
+							*/
+
```
Comment (CL-9a) [inline index.js]: This is just replacing the `forEach` with a `for ... of` so that you can do `break`, right? Could you also clean it up, and possibly make a separate commit / PR for it?

Ideally all that logic would be in a separate file and there'd be tests where we can easily see if anything broke + the time it took, but... this isn't an ideal world and I've been lazy

**Review by CL-9a — CHANGES_REQUESTED** [review]
how do reviews work i'm testing things out
(overall it's good to me :ok_hand: just want to clean up the commented-out stuff to avoid cluttering the already-messy code, and since git tracks the changes, if we want to rollback we can literally roll back a commit, so there's not much point in keeping old code around)

**Comment by SupersonicNK** [comment]
I cleaned out the commented code in two steps (the old code first, then the hastily tested block in a separate commit).

File: index.js (hunk 4 above)
Comment (SupersonicNK) [reply index.js]: That's correct, yeah. I want to test it more thoroughly at some point as I'm still not sure of the performance impact between the two. 
`for ... of` might perform better in the case where there are several extra shade slots, but I don't think it'd be an impact more than negligible still.

**Review by CL-9a — APPROVED** [review]
